src/Compilers/CSharp/Portable/Symbols/AnonymousTypes/AnonymousTypeManager.Templates.cs

```diff
@@ -314,6 +327,15 @@ public void AssignTemplatesNamesAndCompile(MethodCompiler compiler, PEModuleBuil
 
             builder.Free();
 
+            // Ensure all previous sythesized delegates are included so the
+            // types are available for subsequent edit and continue generations.
+            foreach (var key in moduleBeingBuilt.GetPreviousSynthesizedDelegates())
+            {
+                GeneratedNames.ParseSynthesizedDelegateName(key.Name, out var refKinds, out var returnsVoid, out var generation, out var parameterCount);
```

Comment:
>ParseSynthesizedDelegateName [](http://example.com/codeflow?start=31&length=28)

Here we assume that the synthesized name is well-formed. But in GetSynthesizedDelegateMapFromMetadata we read the name from metadata w/o validation. So it may happen that the metadata file is corrupted and we crash here.